src/celeritas/phys/PrimaryGeneratorOptions.hh

```diff
@@ -72,5 +78,21 @@ struct PrimaryGeneratorOptions
     }
 };
 
+//---------------------------------------------------------------------------//
+// FREE FUNCTIONS
+//---------------------------------------------------------------------------//
+
+// Return a distribution for sampling the energy
+std::function<real_type(std::mt19937&)>
+make_energy_sampler(DistributionOptions options);
```

Comment:
Maybe a new type alias `PrimaryGeneratorEngine`? that can be reused in the PG and the new Geant4 PG too.